pynance/textimporter.py

```diff
@@ -0,0 +1,205 @@
+            any object with a read() method.
+            Passed to pandas.read_csv
+        cfd : CsvFileDescription, a description of how the CSV file is to be
+            read and transformed
+        """
+        columns_to_read = cfd.column_map.values()
+
+        # read the dataframe as it is, with only strings
+        # formatting is done later
+        df_as_is = pd.read_csv(filepath_or_buffer=filepath_or_buffer,
+                               dialect=cfd.csv_dialect,
+                               skiprows=cfd.skiprows,
+                               encoding=cfd.encoding,
+                               usecols=columns_to_read,
+                               dtype=str)
+
+        # construct a new DataFrame that matches the definitions
+        new_df = pd.DataFrame()
+
+        # iterate over all the columns that we want to have in the end
+        for new_col_name, new_type in COLUMNS.items():
+
+            if new_col_name in cfd.column_map.keys():
+                old_col_name = cfd.column_map[new_col_name]
+
+                if old_col_name not in df_as_is.columns.values:
+                    raise UnsupportedCsvFormat("Column %s was not found in \
+                                            file" % old_col_name)
+
+                # get the formatter for the required type
+                formatter = cfd.formatters[new_type]
+                try:
+                    # apply the formatter
+                    new_col = df_as_is[old_col_name].apply(formatter)
+                except:
+                    raise UnsupportedCsvFormat("Could not convert content of \
+                                               column %s to %s"
+                                               % (old_col_name, new_type))
+
+                new_df[new_col_name] = new_col
+            else:
+                # some columns are not present in the loaded csv,
+                # fill them with emptyness
+                # but with the correct type
+                # str will still become dtype('object')
+                # because that's how pandas works
+                empty_series = pd.Series(dtype=new_type)
+                new_df[new_col_name] = empty_series
+
+        return new_df
+
+
+class CsvFileDescription():
+    def __init__(self,
+                 column_map,
+                 csv_dialect,
+                 formatters,
+                 skiprows,
+                 encoding):
+        """
+        A description of a specific CSV file design.
+        Typically a definition for a specific bank transaction CSV file
+        would be defined as an instance of this class.
+
+        Parameters
+        ----------
+        column_map : dict,  maps the columns  of the csv file (as keys)
+            to the pynance columns (as values)
+        csv_dialect : csv.Dialect, that describes the files inner syntax
+        formatters : dict(type T, func: string item -> T item) that maps the
```

Comment:
Maybe add a note that `T` is the numpy type?

Reply:
`str` for example is not a numpy type. The whole type thing is not yet very clear. In the end, we want to have the types that are used by pandas - which does not really use `str`, but uses object or `category`-types. It's a broader topic, we that have to discuss. I opened issue #21 for this.

Reply:
Ok. So let's leave it as is.